site/blog/2016/jul/1/sphinx-and-rtd-for-writers.rst

```diff
@@ -0,0 +1,548 @@
+* A builder that outputs all the things changed in the latest version of your code for a changelog
+* An XML builder that outputs a representation of your documents in XML
+* A Man page builder that builds man pages from your documentation
+* JSON builder that outputs your pages as HTML inside of JSON, with some metadata, for embedding dynamically
+
+Trade offs with Sphinx
+----------------------
+
+Every tool has it's issues and limitations.
+I'd like to address some of the major issues with Sphinx,
+so that you can go into using it with eyes wide open.
+
+The biggest issue is that it is originally a *programmer tool*.
+This means that a lot of the designs assume knowledge of code,
+especially for installation and extending the tools.
+Knowledge of the command line is definitely required.
+
+The markup language,
+reStructuredText,
+is also a bit finicky.
+It depends on whitespace for separation of content,
+which is a natural concept for Python programmers,
+but not for most writers.
+
+In general though,
+a lot of the complexity in the language comes from the extensibility and power.
+When compared to something like Markdown,
+reStructuredText can do so much more that it's worth the complexity and somewhat steep learning curve.
+
+Introduction to Read the Docs
+-----------------------------
+
+Read the Docs is a hosting platform for Sphinx documentation.
+It takes the powerful tool that Sphinx is,
+then adds understanding of version control and other powerful features on top.
+
+Supporting Git, Mercurial, and Subversion,
+Read the Docs pulls down the code from your version control,
+then builds and host that documentation for you.
+We'll use GitHub in this example as it's the most commonly used system for accessing code.
+
+To get started,
+you create a Read the Docs account,
+and link your GitHub account.
+Then you select the GitHub repository you'd like to build documentation for,
+at which point the magic happens.
+
+Read the Docs:
+
+* Clones your repository
+* Builds HTML, PDF, and ePub versions of your documentation from your ``master`` branch.
+* Indexes your documentation to allow full-text-search
+* Creates Version objects from each *tag* and *branch* in your repository, allowing you to optionally host those as well
+* Activates a webhook on your repository, so when you push code to any branch, your documentation is rebuilt
+
+Now,
+whenever you commit new code or documentation to your repository,
+your documentation is kept up to date.
+This works with your ``master`` branch,
+as well as any other branches or tags you might have activated documentation for.
+
+Read the Docs has two special versions:
+
+* ``latest`` which maps to the most up to date development version of your software
+* ``stable`` which maps to the latest tagged release of your software
+
+These are version aliases that are meant to maintain stable URL's for your users who want to use the most up to date committed or released software, respectively.
```

Comment:
"These are version aliases that help maintain stable URLs for the most up-to-date commits or for the most stable released version of your software." (don't think you need "respectively")